### ISSUE TITLE
Handle cookie value on R side

### ISSUE THREAD

**Issue by jonthegeek** [shinyworks/cookies#78]
Move the handling that was added in #77 into the R side of the setter. This will require refactoring all setters to use a common value checker, which should almost certainly use S3 methods to prep the values (or possibly just a single {jsonlite} call, I'll have to check).

And then the getters need to match.

Also add tests for that functionality.

**Comment by yogat3ch** [comment]
@jonthegeek,
When you pass a JSON string as the value, Shiny.sendCustomMessage automatically converts to an object on the JS side, that's why I ended up putting it where I did. Does that make sense?

**Comment by jonthegeek** [comment]
@yogat3ch Fair enough! I suspect we may end up with it in both places, ultimately. I'd like to check what you're passing in as the value, and deal with it (both reading and writing) without the user having to think about it too much!

**Comment by yogat3ch** [comment]
The value is `list(a = TRUE, b = TRUE)` ie
```r
cookies_set(
  cookie_name = "test",
  cookie_value = list(a = TRUE, b = TRUE)
)
```

**Comment by jonthegeek** [comment]
Ok, got it. I want to make sure we catch it if the value can't JSONify, and report a clear error on the R side. And then I want the parsing to be clean on the other side.

It's possible more of this already works than I was thinking, but part of the refactor will involve making it easier to see what is and isn't working.

More in January!

**Comment by yogat3ch** [comment]
@jonthegeek, that makes sense regarding catching if its not JSON-able. Ok, sounds good!
